drivers/adc/adc_ads1119.c

```diff
@@ -0,0 +1,528 @@
+{
+	const struct ads1119_config *config = dev->config;
+
+	return i2c_reg_read_byte(config->i2c_dev, config->i2c_dt_device_addr,
+				 ADS1119_CMD_READ_REG | reg_addr, reg_val);
+}
+
+static int ads1119_write_reg(const struct device *dev, uint8_t reg)
+{
+	const struct ads1119_config *config = dev->config;
+
+	return i2c_reg_write_byte(config->i2c_dev, config->i2c_dt_device_addr,
+				  ADS1119_CMD_WRITE_REG, reg);
+}
+
+static inline int ads1119_acq_time_to_dr(const struct device *dev,
+					 uint16_t acq_time)
+{
+	struct ads1119_data *data = dev->data;
+	int odr = -EINVAL;
+	uint16_t acq_value = ADC_ACQ_TIME_VALUE(acq_time);
+
+	if (acq_time != ADC_ACQ_TIME_DEFAULT &&
+	    ADC_ACQ_TIME_UNIT(acq_time) != ADC_ACQ_TIME_MILLISECONDS) {
+		return -EINVAL;
+	}
+
+	switch (acq_value) {
+	case ADC_ACQ_TIME_VALUE(ADC_ACQ_TIME_DEFAULT):
+	case 1000 / 20:
+		odr = ADS1119_CONFIG_DR_20;
+		data->ready_time = K_MSEC(1000 / 20);
+		break;
+	case 1000 / 90:
+		odr = ADS1119_CONFIG_DR_90;
+		data->ready_time = K_MSEC(1000 / 90);
+		break;
+	case 1000 / 330:
+		odr = ADS1119_CONFIG_DR_330;
+		data->ready_time = K_MSEC(1000 / 330);
+		break;
+	case 1000 / 1000:
+		odr = ADS1119_CONFIG_DR_1000;
+		data->ready_time = K_MSEC(1000 / 1000);
+		break;
+	default:
+		break;
+	}
+
+	/* As per datasheet acquisition time is a bit longer wait a bit more
+	 * to ensure data ready at first try
+	 */
+	data->ready_time.ticks += K_USEC(10).ticks;
+
+	return odr;
+}
+
+static int ads1119_send_start_read(const struct device *dev)
+{
+	const struct ads1119_config *config = dev->config;
+	const uint8_t cmd = ADS1119_CMD_START_SYNC;
+
+	return i2c_write(config->i2c_dev, &cmd, sizeof(cmd), config->i2c_dt_device_addr);
+}
+
+static int ads1119_wait_data_ready(const struct device *dev)
+{
+	int rc;
+	struct ads1119_data *data = dev->data;
+
+	k_sleep(data->ready_time);
+	uint8_t status = 0;
+
+	rc = ads1119_read_reg(dev, ADS1119_REG_STATUS, &status);
+	if (rc != 0) {
+		return rc;
+	}
+
+	while ((status & ADS1119_STATUS_MASK_READY)
+	       >> ADS1119_STATUS_SHIFT_READY == 0) {
+
+		k_sleep(K_USEC(100));
+		rc = ads1119_read_reg(dev, ADS1119_REG_STATUS, &status);
+		if (rc != 0) {
+			return rc;
+		}
+	}
+
+	if ((status & ADS1119_STATUS_MASK_READY)
+	    >> ADS1119_STATUS_SHIFT_READY == 1) {
+		rc = 0;
+	} else {
+		rc = -ETIMEDOUT;
+	}
+
+	return rc;
+}
+
+static int ads1119_read_sample(const struct device *dev, uint16_t *buff)
+{
+	int res;
+	uint8_t rx_bytes[2];
+	const struct ads1119_config *config = dev->config;
+	const uint8_t cmd = ADS1119_CMD_READ_DATA;
+
+	res = i2c_write_read(config->i2c_dev, config->i2c_dt_device_addr,
+			     &cmd, sizeof(cmd),
+			     rx_bytes, sizeof(rx_bytes));
+
+	*buff = sys_get_be16(rx_bytes);
+	return res;
+}
+
+static int ads1119_channel_setup(const struct device *dev,
+				 const struct adc_channel_cfg *channel_cfg)
+{
+	uint8_t config = 0;
+	int dr = 0;
+
+	if (channel_cfg->channel_id != 0) {
```

Comment:
"_Only one channel is supported..._"

Do you mean only the channel 0 is supported?

Reply:
Yes, but the input of the channel is configurable